data-prepper-plugins/opensearch-source/build.gradle

```diff
@@ -0,0 +1,17 @@
+/*
+  * Copyright OpenSearch Contributors
+  * SPDX-License-Identifier: Apache-2.0
+  */
+dependencies {
+    implementation project(path: ':data-prepper-api')
+    implementation 'com.fasterxml.jackson.core:jackson-databind'
+    implementation 'com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.14.2'
+    implementation 'software.amazon.awssdk:s3'
+    implementation 'software.amazon.awssdk:sts'
+    implementation project(path: ':data-prepper-core')
```

Comment:
You should not have a dependency on data-prepper-core. That will result in circular dependency. You do not need the annotation for `@DataPrepperDurationDeserializer` on any Durations. That will be handled by data-prepper-core when data prepper starts up.

Reply:
Thank You @graytaylor0  for your reviews and comments . The changes are incorporated.

Reply:
@rajeshLovesToCode , I still see the dependency on `data-prepper-core` here.